README.md

```diff
@@ -233,14 +233,164 @@ mstorage.info("<media_id>", new CompletionHandler<MediaInfo>() {
 });
 ```
 
+### Attach media metadata
+You can define your original metadata as a 'user metadata'.  
+Existing metadata value for the same key will be overwritten. Up to 10 user metadata can be attached to a media data. 
+
+```java
+Map<String, String> userMeta = new HashMap<String, String>();
+userMeta.put("user.<key1>", "<value1>");
+userMeta.put("user.<key2>", "<value2>");
+mstorage.addMeta("<media_id>", userMeta, new CompletionHandler<Object>() {
+    @Override
+    public void onCompleted(Object result) {
+        // Do something.
+        // "result" may be empty.
+    }
+
+    @Override
+    public void onThrowable(Throwable t) {
+        // Something wrong happened.
+    }
+});
+```
+
 ### Get media metadata
+* All
 ```java
 mstorage.meta("<media_id>", new CompletionHandler<MediaMeta>() {
     @Override
     public void onCompleted(MediaMeta mediaMeta) {
         Map<String, String> exif = mediaMeta.getExif();
         Map<String, String> gpano = mediaMeta.getGpano();
+        Map<String, String> userMeta = mediaMeta.getUserMeta();
+        // Do something.
+    }
+
+    @Override
+    public void onThrowable(Throwable t) {
+        // Something wrong happened.
+    }
+});
+```
+
+* Exif
+```java
+mstorage.meta("<media_id>", "exif", new CompletionHandler<MediaMeta>() {
+    @Override
+    public void onCompleted(Map<String, String> exif) {
+        // Do something.
+    }
+
+    @Override
+    public void onThrowable(Throwable t) {
+        // Something wrong happened.
+    }
+});
+```
+
+* Google Photo Sphere XMP
+```java
+mstorage.meta("<media_id>", "gpano", new CompletionHandler<MediaMeta>() {
+    @Override
+    public void onCompleted(Map<String, String> exif) {
+        // Do something.
+    }
+
+    @Override
+    public void onThrowable(Throwable t) {
+        // Something wrong happened.
+    }
+});
+```
+
+* User metadata (all)
+```java
+mstorage.meta("<media_id>", "user", new CompletionHandler<MediaMeta>() {
+    @Override
+    public void onCompleted(Map<String, String> userMeta) {
+        // Do something.
+    }
+
+    @Override
+    public void onThrowable(Throwable t) {
+        // Something wrong happened.
+    }
+});
+```
+
+* User metadata (with a key)
+```java
+mstorage.meta("<media_id>", "user.<key>", new CompletionHandler<MediaMeta>() {
+    @Override
+    public void onCompleted(Map<String, String> userMeta) {
+        // Do something.
+    }
+
+    @Override
+    public void onThrowable(Throwable t) {
+        // Something wrong happened.
+    }
+});
+```
+
+* Search
```

Comment:
This method returns MediaList so it's different from other meta methods which return all or some of metadata.
Search should be listed in "Get media ids" section.